### PR TITLE
Remove support for GopherJS to unblock newer Go versions

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

The leaves the GopherJS support in cmd/fyne intact given that the plan is to move that tool out of the repository for this release.

For #3242

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Comment by andydotxyz** [comment]
The CI seems pretty unhappy - is this a false negative?

```
# fyne.io/fyne/v2/internal/async
Error: ../../internal/async/queue_canvasobject.go:13:36: undefined: itemCanvasObject
Error: ../../internal/async/queue_canvasobject.go:17:10: undefined: CanvasObjectQueue
Error: ../../internal/async/queue_canvasobject.go:18:36: undefined: itemCanvasObject
Error: ../../internal/async/queue_canvasobject.go:22:22: undefined: itemCanvasObject
Error: ../../internal/async/queue_canvasobject.go:24:10: undefined: loadCanvasObjectItem
Error: ../../internal/async/queue_canvasobject.go:25:14: undefined: loadCanvasObjectItem
Error: ../../internal/async/queue_canvasobject.go:26:6: undefined: loadCanvasObjectItem
Error: ../../internal/async/queue_canvasobject.go:[28](https://github.com/fyne-io/fyne/actions/runs/7330832004/job/19962560298?pr=4486#step:5:29):8: undefined: casCanvasObjectItem
Error: ../../internal/async/queue_canvasobject.go:42:10: undefined: CanvasObjectQueue
Error: ../../internal/async/queue_canvasobject.go:67:10: undefined: CanvasObjectQueue
Error: ../../internal/async/queue_canvasobject.go:28:8: too many errors
Error: Process completed with exit code 1.
```

**Comment by Jacalz** [comment]
Was an error my part. Forgot to update the generator script.

**Comment by andydotxyz** [comment]
With this we have 2 build targets for web - "web" and "wasm". Should we not just default down to "web" which makes the wasm build (as that's an implementation detail).

**Comment by Jacalz** [comment]
The `web` vs `wasm` build targets are never used on a build tag level. The `cmd/fyne` tool just built a GopherJS + WASM when using `web` as I understand it. The change for that will be in the migrated tool repo. I don't think there is anything to change here. I'll make sure to get the migrated cmd tool updated once this has landed.

**Comment by andydotxyz** [comment]
Hmm, I agree the work is done by the package tool, but the build flags still are important.
If you can remove all the `js` tags in this commit it will give a better understanding of what other files are impacted and whether anything more needs to change.

**Comment by Jacalz** [comment]
Good point. Did the same thing in glfw-js just now. Will do that here as well.

File: internal/async/queue_pure_canvasobject.go
Comment (Jacalz) [inline internal/async/queue_pure_canvasobject.go:1]: FYI: I deleted this file as it isn't necessary any more. WASM is a single-threaded architecture but the Go scheduler can still schedule gorountines to run concurrently and race conditions can still happen. I double checked by asking in the #go-wasm channel on Gopher's Slack. The old code was either incorrect or GopherJS related (I don't know which but I think the former).

**Review by andydotxyz — APPROVED** [review]
Thanks

**Comment by Jacalz** [comment]
Very exciting. It will be nice seeing some modernisation in the code base :)

**Pull request merged**